onesignal-settings.php

```diff
@@ -253,8 +253,8 @@ public static function get_onesignal_settings() {
         }
       }
     }
-
-    return apply_filters( 'onesignal_get_settings', $onesignal_wp_settings );
+    $onesignal_settings_values_without_slashes = stripslashes_deep($onesignal_wp_settings);
```

Comment:
This looks to be running `stripslashes_deep` on all values in the `onesignal_wp_settings` hash. There are some fields such as `utm_additional_url_params` where slashes are expected however so I think we should only run this for specific fields.

Reply:
## The Problem:
The customer adds apostrophe to any field in the OneSignal settings:
<img width="1300" alt="screen_shot_2021-01-14_at_2 32 33_pm" src="https://user-images.githubusercontent.com/11739227/104657768-2cf8ab00-5687-11eb-9e19-4512b60479d8.png">

Upon save, a slash gets added before the apostrophes. This happens on every save which leads to a buildup of slashes.

<img width="354" alt="Screen Shot 2021-01-14 at 2 33 09 PM" src="https://user-images.githubusercontent.com/11739227/104657761-2a965100-5687-11eb-95bc-52e475ad4f33.png">

These slashes show up on the prompt and the apostrophe gets translated to HTML version:

<img width="602" alt="Screen Shot 2021-01-14 at 2 33 35 PM" src="https://user-images.githubusercontent.com/11739227/104657756-2702ca00-5687-11eb-8b1a-8025ef0e6ff5.png">

## Solution

The following two methods are needed together in order for the fix to work:

`stripslashes_deep` was added to remove slashes automatically added upon saving with an apostrophe. It does not remove slashes added purposely e.g: in UTM params. a side-effect of this workaround is that the apostrophe is converted to HTML version: `&#039` which is why the next function is also needed.

`decode_entities` was added because without it, apostrophes would show up as the HTML version: `&#039;`. This function was previously used elsewhere in the plugin.

Reply:
Thanks for checking and explaining in detail. This is great documentation if we need to look back on what this is fixing if we ever want to change this implementation in the future for any reason.